### PR TITLE
Revendor Microsoft/hcsshim and go-winio, plus container/containerd

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <john.howard@microsoft.com>

Bumps hcsshim to 0.6.14 and go-winio to 0.4.10

### CODE REVIEW

**Comment by thaJeztah** [comment]
Full diff;

- https://github.com/Microsoft/hcsshim/compare/v0.6.12...v0.6.14
  - https://github.com/Microsoft/hcsshim/pull/272 Changes to support registry modification in containers
  - https://github.com/Microsoft/hcsshim/pull/281 v0.6: Limit parallel starts
  - https://github.com/Microsoft/hcsshim/pull/283 Fix fileattributes in the v0.6-dev branch to match current go-winio


- https://github.com/Microsoft/go-winio/compare/v0.4.9...v0.4.10
  - https://github.com/Microsoft/go-winio/pull/88 Update FileAttributes size to work across all platforms

**Comment by thaJeztah** [comment]
@jhowardmsft build is failing;

```
16:46:12 # github.com/docker/docker/vendor/github.com/containerd/containerd/archive
16:46:12 ..\..\vendor\github.com\containerd\containerd\archive\tar_windows.go:262:27: cannot use uintptr(attr) (type uintptr) as type uint32 in assignment
16:46:23 
16:46:23 ERROR: make.ps1 failed:
16:46:23 Failed to compile daemon
```

**Comment by jhowardmsft** [comment]
@thaJeztah Hold fire on this - there's a commit missing. Changing to WIP

**Comment by jhowardmsft** [comment]
@thaJeztah It's because containerd/containerd needs bumping too due to it's revendoring. Dependent on https://github.com/containerd/containerd/pull/2557 and a revendor of containerd into moby/moby once that is merged.

**Comment by thaJeztah** [comment]
@jhowardmsft https://github.com/containerd/containerd/pull/2557 is merged now 👍

**Comment by jhowardmsft** [comment]
@thaJeztah Yup, saw. Bumped containerd, fingers crossed all should be good now.

File: vendor.conf [hunk 1]
```diff
@@ -114,7 +114,7 @@ github.com/googleapis/gax-go v2.0.0
 google.golang.org/genproto 694d95ba50e67b2e363f3483057db5d4910c18f9
 
 # containerd
-github.com/containerd/containerd v1.2.0-beta.0
+github.com/containerd/containerd 3f42445e38d1081f4b8c3b8d7d1ed1860198ed7a
```
Comment (thaJeztah) [inline vendor.conf]: ping @dmcgowan should the upstream change be included in the 1.2.x branch?

Comment (dmcgowan) [reply vendor.conf]: The 1.2.x branch won't get created until first rc, this is fine

Comment (thaJeztah) [reply vendor.conf]: Ah, perfect; thanks!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by dmcgowan** [comment]
LGTM

**Comment by thaJeztah** [comment]
Janky and Experimental failing on; https://jenkins.dockerproject.org/job/Docker-PRs/50499/console

```
20:34:41 --- FAIL: TestAuthZPluginEnsureLoadImportWorking (1.38s)
20:34:41 	daemon.go:289: [d69ba33f8b5fc] waiting for daemon to start
20:34:41 	daemon.go:321: [d69ba33f8b5fc] daemon started
20:34:41 	authz_plugin_test.go:380: assertion failed: error is not nil: unexpected EOF
20:34:41 	daemon.go:279: [d69ba33f8b5fc] exiting daemon
20:34:41 	main_test.go:57: Error while stopping the daemon d69ba33f8b5fc : exit 
```

not sure I've seen that one before; I'll restart but could be legit

**Comment by thaJeztah** [comment]
Moving back to code-review; seeing the same failure on Janky and Experimental again, so may be legit; possibly caused by the containerd bump

@dmcgowan ^^

**Comment by jhowardmsft** [comment]
@crosbymichael As mentioned offline. Any chance you can see what's up with the timeout change. That looks like the only relevant thing here.

**Comment by crosbymichael** [comment]
60 seconds was the prior default, I guess we know why now... 60 seconds to connect to a unix socket, waaaaat

**Comment by jhowardmsft** [comment]
@crosbymichael Thanks. Yes, pushed a fix. @thaJeztah Hopefully CI will pass now.

**Comment by crosbymichael** [comment]
It looks like this has to do with the tar changes that came in.

**Comment by crosbymichael** [comment]
Your vendor has removed archive/tar code in it.  Maybe it is a failed vndr run but I don't think those should be removed and that will probably resolve the issue

**Comment by jhowardmsft** [comment]
Yeah, I saw that, and was surprised when `vndr` did that. However, the vendor CI job passes and matches my vndr run. I can't explain why vndr has dropped it (clearly something previously referencing it no longer does), or why it should be causing this failure. @thaJeztah Have you any clues here? This is mind boggling.

**Comment by thaJeztah** [comment]
> Yeah, I saw that, and was surprised when vndr did that. However, the vendor CI job passes and matches my vndr run

Oh! I know... You must have done a _full_ `vndr`, but using the current "master" version of vndr, not the version that we pinned to

The `vndr/archive` is a manual vendor, and the current "master" of `vndr` wipes the `vendor` directory before doing the actual vendoring. See https://github.com/LK4D4/vndr/issues/72 (introduced in https://github.com/LK4D4/vndr/pull/65)

To work around this, either;

- only run `vndr` for the dependencies that were updated (`vndr github.com/Microsoft/hcsshim`, `vndr github.com/Microsoft/go-winio`)
- use the pinned version of `vndr`: https://github.com/moby/moby/blob/49ad3d69a7780e54ace50290d0a1a55c1182a947/hack/dockerfile/install/vndr.installer#L3

**Comment by jhowardmsft** [comment]
@thaJeztah Thanks, that explains it. I undid that part. Hopefully Janky/Experimental pass now.

**Comment by thaJeztah** [comment]
Thanks! Can you squash that commit with the one that removed it?

**Comment by jhowardmsft** [comment]
> Thanks! Can you squash that commit with the one that removed it?

Done

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by jhowardmsft** [comment]
@thaJeztah Good to merge now?

**Review by johnstep — APPROVED** [review]

**Comment by thaJeztah** [comment]
all green; merging

**Pull request merged**